templates/terraform/examples/service_directory_endpoint_basic.tf.erb

```diff
@@ -0,0 +1,25 @@
+resource "google_service_directory_namespace" "<%= ctx[:primary_resource_id] %>" {
+  provider     = google-beta
+  namespace_id = "<%= ctx[:vars]["namespace_id"] %>"
+  location     = "us-central1"
+}
+
+resource "google_service_directory_service" "<%= ctx[:primary_resource_id] %>" {
+  provider   = google-beta
+  service_id = "<%= ctx[:vars]["service_id"] %>"
+  namespace  = google_service_directory_namespace.<%= ctx[:primary_resource_id] %>.name
```

Comment:
Can these refer to the id field instead of the name field? https://github.com/terraform-providers/terraform-provider-google/wiki/Developer-Best-Practices#self-links-names-and-ids

Reply:
Thanks for flagging. I always assumed `id` was private to Terraform and shouldn't be used for interpolation. Now I see why `id_format`s were changed to be like `name`s or `self_link`s.

Reply:
Of course! You're right that we used to do that. We decided that for 3.0 we wanted to try to change things so that they could get used as references, so users don't have to know for a given resource whether it's self_link, name, or something else that they should be interpolating on.

Reply:
Thanks. Should be fixed on the main resources now. It's still referencing `name`s on the IAM resources, as it seems to be an issue referencing the `id`s.